src/Microsoft.AspNet.Identity/RoleManager.cs

```diff
@@ -286,7 +286,7 @@ public virtual string NormalizeKey(string key)
             ThrowIfDisposed();
             await Store.SetRoleNameAsync(role, name, CancellationToken);
             await UpdateNormalizedRoleNameAsync(role);
-            return await LogResultAsync(IdentityResult.Success, role);
+            return await Task.FromResult(IdentityResult.Success).WithLoggingAsync(this, role);
```

Comment:
There is just one instance where we return IdentityResult instead of Task in RoleManager so I did not add that. I'll add that change